tests/widget/widget.js

```diff
@@ -110,4 +112,85 @@ describe( 'Widget', () => {
 		sinon.assert.notCalled( focusSpy );
 		expect( getModelData( document ) ).to.equal( '[<widget></widget>]' );
 	} );
+
+	describe( 'delete and backspace handling', () => {
```

Comment:
There are no tests for many potential false positive cases like what if the widget was selected in the first place and is preceded by another one, what if it's preceded by a paragraph with text or paragraph with an inline object, or what if we have inline object preceded by a text. It'd be easier if the probing mechanism was exposed. E.g. as I commented there – it's unclear for me (cause I don't know the API in fact :D) what's nodeBefore/After behaviour in case of text nodes. What if we have something like: `<p>xx[]<obj></obj></p>` case and we press backspace? The focus will be moved between "x[x]" and what's the `nodeAfter` then?